docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+
+In addition, we can notice that the constraint of having both peers connected to the server at the same time in order to make the exchange doesn't provide any substantial gain. On the contrary, it's fine to have both peers simply depositing and retrieving their payload as they please, as long as it is done in a consistent way. The retrieval is the blocking part, which can be either implemented with a blocking request or non-blocking one, returning a "not ready" status when the payload is not available. This is the approach used here in order to avoid dealing with blocking requests altogether (although it would be valid in this context since retrying after a network error is mostly equivalent to retrying after a "not ready" status).
+
+Since the invite transport is being redesigned, we're including two more changes to this RFC. First, we need to add a reason for an attempt being cancelled by a peer. This is useful for the front-end application to properly communicate to the user what actually happened. This reason did not exist in v2 and has to be added in v3. More precisely, we can notice that the semantics of "attempts" is absent from the v2, even though the process of creating a secure channel between peers can still be cancelled and/or reset. Without dedicated semantics, it is hard to properly manage edge cases such as two devices for the same user trying to create a secure channel at the same time. For this reason, a new attempt semantics is defined here such that each attempt is properly identified and dealt with separately.
+
+The second change is related to the concept of greeters. In Parsec v2, only the administrator that created an invitation is able to greet the invited user. This is something we want to change to open the way to easier invitations, where any administrator can greet a user that has been invited. This may be configurable of course, but the transport should allow for multiple greeters per invitation. Note that this is already the case for shamir invitations, where the greeters are the shamir recipients. For this reason, the new design can draw inspiration from what was done during the shamir implementation. More precisely, each invitation will have its corresponding set of greeters which is different for each invitation type:
```

Comment:
```suggestion
The second change is related to the concept of greeters. In Parsec v2, only the administrator that created an invitation is able to greet the invited user. This is something we want to change to open the way to easier invitations, where any administrator can greet a user that has been invited. This may be configurable of course, but the transport should allow for multiple greeters per invitation. Note that this is already the case for Shamir invitations, where the greeters are the Shamir recipients. For this reason, the new design can draw inspiration from what was done during the Shamir implementation. More precisely, each invitation will have its corresponding set of greeters which is different for each invitation type:
```

Reply:
d36e8a3440bf5a5cb645e168de63108650bc36bf